packages/shared_preferences/shared_preferences_android/lib/src/shared_preferences_async_android.dart

```diff
@@ -130,7 +133,22 @@ base class SharedPreferencesAsyncAndroid
     final SharedPreferencesPigeonOptions pigeonOptions =
         _convertOptionsToPigeonOptions(options);
     final SharedPreferencesAsyncApi api = _getApiForBackend(pigeonOptions);
-    return api.setStringList(key, value, pigeonOptions);
+    final String stringValue = '$jsonListPrefix${jsonEncode(value)}';
+    return api.setString(key, stringValue, pigeonOptions);
+  }
+
+  /// Adds a `List<String>` to preferences using platform encoding to test
+  /// moving from platform encoding to json encoding.
+  @visibleForTesting
+  Future<void> setStringListLegacyForTesting(
```

Comment:
Generally the less test-only code in production code files the better; could we just expose `_getApiForBackend` for testing instead?

Reply:
I'm not sure I understand this suggestion, are you saying I should add an option in that method that sets which version of setString is used?